rta/__init__.py

```diff
@@ -39,6 +45,116 @@ def get_ttp_names(os_types=None):
     return names
 
 
+def get_available_tests(print_list=False, os_filter=None):
+    test_names = []
+    test_metadata = []
+
+    for file in list(CURRENT_DIR_PATH.rglob("*.py")):
+        if file.stem not in ("common", "main") and not file.stem.startswith("_"):
+            module = importlib.import_module(f"rta.{file.stem}")
+            if os_filter and os_filter not in module.PLATFORMS:
+                continue
+            test_names.append(file.stem)
+            test_metadata.append({"name": file.stem, "path": file, "platforms": module.PLATFORMS})
+
+    if print_list:
+        longest_test_name = len(max(test_names, key=len))
+        print("Printing available tests")
+        header = f"{'name':{longest_test_name}} | {'platforms':<30}"
+        print(header)
+        print("=" * len(header))
+        [print(f"{test['name']:<{longest_test_name}} | {', '.join(test['platforms'])}") for test in test_metadata]
+
+    return test_names, test_metadata
+
+
+def get_rta_triggered_rules() -> dict:
+    # get all rtas and the rules they cover
+    all_rtas = {"windows": [], "linux": [], "macos": []}
+    for file in list(CURRENT_DIR_PATH.rglob("*.py")):
+        if file.stem not in ("common", "main") and not file.stem.startswith("_"):
+            module = importlib.import_module(f"rta.{file.stem}")
+            for platform in module.PLATFORMS:
+                all_rtas[platform].extend(module.TRIGGERED_RULES.get("SIEM"))
+    return all_rtas
+
+
+def get_os_list(rule) -> list:
+    os_list = []
+    if rule.contents.metadata.os_type_list:
+        os_list = [r.lower() for r in rule.contents.metadata.os_list]
+    elif rule.contents.data.tags:
+        tags = [t.lower() for t in rule.contents.data.tags]
+        core_os = ["windows", "linux", "macos"]
+        for os_type in core_os:
+            if os_type in tags:
+                os_list.append(os_type)
+    return os_list
+
+
+def build_coverage_map(all_rtas, all_rules) -> dict:
+    # get the rules that are not covered by each rta
```

Comment:
can you print this out. I am not clear on the intended use for this

Reply:
It compares triggered rules in the rtas against all rules to see which rules are covered by an RTA